### ISSUE TITLE
Findbugs fixes

### ISSUE THREAD

**Issue by ahaarrestad** [OfficeDev/ews-java-api#200]
As of the today's source, findbugs found the following
8 scary
3 troubling
2 of concern

at least the 8 scary issues must be fixed!

**Comment by vboctor** [comment]
Makes sense.  Is there also a way to integrate this tool in Travis CI to keep the code clean once we make it clean?

**Comment by ahaarrestad** [comment]
We can use finbugs mojo in the reporting part of the pom.xml, and this will fail the build during verify if any bugs are found.

**Comment by ahaarrestad** [comment]
fixed a few issues. could need some input on the following scary that is harder to fix


          timestamp = String.format(WSSecurityBasedCredentials.wsuTimeStampFormat, utcNow, utcNow);

    protected static final String wsuTimeStampFormat =
        "<wsu:Timestamp>" +
            "<wsu:Created>{0:yyyy'-'MM'-'dd'T'HH':'mm':'ss'Z'}</wsu:Created>" +
            "<wsu:Expires>{1:yyyy'-'MM'-'dd'T'HH':'mm':'ss'Z'}</wsu:Expires>" +
            "</wsu:Timestamp>";
